sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -245,6 +245,19 @@ case class Distinct(partial: Boolean, child: SparkPlan) extends UnaryNode {
   }
 }
 
+/**
+ * :: DeveloperApi ::
+ * Return a new RDD that has exactly `numPartitions` partitions.
+ */
+@DeveloperApi
+case class Coalesce(numPartitions: Int, shuffle: Boolean, child: SparkPlan) extends UnaryNode {
```

Comment:
@rxin Should I rename this to `Repartition`? There are a lot of conflicts coming from `catalyst` and `sql`. In fact, the `Coalesce` function in catalyst fits it's usage, which is to `combine (elements) in a mass or whole.`. Here, we are basically repartitioning the dataset. `Coalesce` with a higher number of partitions sounds weird. Also it might be weird to have two different types of `Coalesce`. What do you think?